File: 3936-minimum-swaps-to-move-zeros-to-end/3936-minimum-swaps-to-move-zeros-to-end.py
class Solution:
    def minimumSwaps(self, nums: list[int]) -> int:
        count0 = nums.count(0)
        if count0 == 0 or count0 == len(nums):
            return 0
        zero_place = nums[-count0:]
        misplacedNums = len(zero_place) - zero_place.count(0)
        return misplacedNums

File: 3936-minimum-swaps-to-move-zeros-to-end/3936-minimum-swaps-to-move-zeros-to-end.cpp
class Solution {
public:
    int minimumSwaps(vector<int>& nums) {
        int total_zeros = count(nums.begin(), nums.end(), 0);

        if (total_zeros == 0 || total_zeros == nums.size())
            return 0;
        
        int misplaced_non_zero = 0;
        int n = nums.size();
        for (int i = n - total_zeros; i < n; i++) {
            if (nums[i] != 0) {
                misplaced_non_zero++;
            }
        }
        return misplaced_non_zero;
    }
};